ChatBot/ChatBot/ChattingRoomView/ViewController/ChattingRoomViewController.swift

```diff
@@ -0,0 +1,277 @@
+        let layout = UICollectionViewCompositionalLayout.list(using: configuration)
+        return layout
+    }
+    
+    private func configureDataSource() {
+        let cellRegistration = UICollectionView.CellRegistration<MessageCell, Message> { (cell, indexPath, item) in
+            cell.message = item
+        }
+        
+        dataSource = DataSource(collectionView: chattingRoomView) { (chattingRoomView, indexPath, itemIdentifier) -> MessageCell? in
+            let cell = chattingRoomView.dequeueConfiguredReusableCell(using: cellRegistration, for: indexPath, item: itemIdentifier.message)
+            return cell
+        }
+        
+        snapshot.appendSections(Section.allCases)
+        dataSource.apply(snapshot, animatingDifferences: true)
+    }
+    
+    private func configureDelegate() {
+        messageTextView.delegate = self
+    }
+}
+
+// MARK: DataSource Controls
+extension ChattingRoomViewController {
+    private enum Section: CaseIterable {
+        case main
+    }
+    
+    private struct Item: Hashable {
+        let id: String
+        let message: Message
+        
+        init(id: String = UUID().uuidString, message: Message) {
+            self.id = id
+            self.message = message
+        }
+        
+        func hash(into hasher: inout Hasher) {
+            hasher.combine(id)
+        }
+        
+        static func == (lhs: ChattingRoomViewController.Item, rhs: ChattingRoomViewController.Item) -> Bool {
+            return lhs.id == rhs.id
+        }
+    }
+    
+    private func updateDataSource(with messages: [Message?]) {
+        let items: [Item] = messages.compactMap { message in
+            guard let message = message else { return nil }
+            return Item(message: message)
+        }
+        
+        snapshot.appendItems(items)
+        dataSource.apply(snapshot, animatingDifferences: true)
+        
+        scrollToBottom(itemsCount: snapshot.numberOfItems, sectionsCount: snapshot.numberOfSections)
+    }
+    
+    private func updateDataSource(with message: Message) {
+        let item = Item(message: message)
+        
+        snapshot.appendItems([item])
+        dataSource.apply(snapshot, animatingDifferences: true)
+        
+        scrollToBottom(itemsCount: snapshot.numberOfItems, sectionsCount: snapshot.numberOfSections)
+    }
+    
+    private func reloadCurrentDataSource(with message: Message) {
+        guard let lastItem = snapshot.itemIdentifiers.last else { return }
+        let newItem = Item(message: message)
+        snapshot.deleteItems([lastItem])
+        snapshot.appendItems([newItem])
+        dataSource.apply(snapshot, animatingDifferences: true)
+        messageTextView.isEditable = true
+    }
+    
+    private func scrollToBottom(itemsCount: Int, sectionsCount: Int, at position: UICollectionView.ScrollPosition = .bottom) {
+        guard itemsCount > 0,
+              sectionsCount > 0
+        else { return }
```

Comment:
guard ~ else 까지 한줄로 처리하고 return 을 확실하게 볼 수 있게 다음과 같이 개행 처리해주는게 좋은 것 같아요.
예외 처리가 되는 부분을 잘 볼 수 있어서요!
```suggestion
        guard itemsCount > 0, sectionCount > 0 else {
            return
        }

Reply:
다음 step에 반영하겠습니다!